dynamo/tools/clustering.py

```diff
@@ -174,36 +173,61 @@ def hdbscan(
+    For other community detection general parameters,
+    please refer to ``dynamo's`` :py:meth:`~dynamo.tl.cluster_community` function.
+    "The Leiden algorithm is an improvement of the Louvain algorithm.
+    The Leiden algorithm consists of three phases:
+    (1) local moving of nodes,
+    (2) refinement of the partition,
+    (3) aggregation of the network based on the refined partition,
+    using the non-refined partition to create an initial partition for the aggregate network." - cdlib
 
-    Parameters
-    ----------
-    weight :
-         weights of edges. Can be either an iterable or an edge attribute. Default None
-    initial_membership : optional
-        list of int Initial membership for the partition. If None then defaults to a singleton partition. Default None, by default None
+    Args:
+        adata: an adata object
+        resolution: the resolution of the clustering that determines the level of detail in the clustering process.
+            An increase in this value will result in the generation of a greater number of clusters.
+        use_weight: whether to use the weight of the edges in the clustering process. Default True
+        weight: weights of edges. Can be either an iterable(list of double) or an edge attribute. Default None
+        initial_membership: list of int Initial membership for the partition.
+            If None then defaults to a singleton partition.
+        adj_matrix: the adjacency matrix to use for clustering. Default None
+        adj_matrix_key: adj_matrix_key in adata.obsp used for clustering. Default None
+        result_key: the key to use for the clustering results. Default None
+        layer: the adata layer where cluster algorithms will work on, by default None
+        obsm_key: the key of the obsm to use for a function of neighbors. Default None
+        selected_cluster_subset: a tuple of 2 elements (cluster_key, allowed_clusters).
+            filtering cells in adata based on cluster_key in adata.obs and only reserve cells in the allowed clusters.
+        selected_cell_subset: cluster only a subset of cells in adata, by default None. Default None
+        directed: whether the graph is directed. Default False
+        copy: return a copy instead of writing to adata. Default False
+        **kwargs: additional arguments to pass to the clustering function.
 
+    Returns:
+        adata: an updated AnnData object with the clustering updated.
+            Each result_key corresponds to two newly added columns, both .obs and .uns attribute.
+            These columns contain either the clustering results or the probability of each cell belonging to a cluster.
```

Comment:
Do we ever return the probability of each cell belong to a cluster? I think we only return the cluster id for each cell. Please make sure your description is precise again